x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/AgentConfigurationCreateEdit/SettingsPage/SettingFormRow.tsx

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import {
+  EuiFormRow,
+  EuiFieldText,
+  EuiFieldNumber,
+  EuiDescribedFormGroup,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiCode,
+  EuiSpacer,
+  EuiIconTip
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import {
+  amountAndUnitToString,
+  amountAndUnitToObject
+} from '../../../../../../../../../../plugins/apm/common/runtime_types/agent_configuration/amount_and_unit';
+import {
+  SettingDefinition,
+  isValid
+} from '../../../../../../../../../../plugins/apm/common/runtime_types/agent_configuration/config_setting_definitions';
+import { SelectWithPlaceholder } from '../../../../../shared/SelectWithPlaceholder';
+
+function FormRow({
+  setting,
+  value,
+  onChange
+}: {
+  setting: SettingDefinition;
+  value?: string;
+  onChange: (key: string, value: string) => void;
+}) {
+  const defaultInputPlaceholder = i18n.translate(
+    'apm.agentConfig.defaultInputPlaceholder',
+    {
+      defaultMessage: 'Set {settingsLabel}',
+      values: { settingsLabel: setting.label }
+    }
+  );
+
+  switch (setting.type) {
+    case 'text': {
+      return (
+        <EuiFieldText
+          placeholder={setting.placeholder || defaultInputPlaceholder}
+          value={value || ''}
+          onChange={e => onChange(setting.key, e.target.value)}
+        />
+      );
+    }
+
+    case 'integer': {
+      return (
+        <EuiFieldNumber
+          placeholder={setting.placeholder || defaultInputPlaceholder}
+          value={(value as any) || ''}
+          min={setting.min}
+          max={setting.max}
+          onChange={e => onChange(setting.key, e.target.value)}
+        />
+      );
+    }
+
+    case 'select': {
+      return (
+        <SelectWithPlaceholder
+          placeholder={setting.placeholder}
+          options={setting.options}
+          value={value}
+          onChange={e => onChange(setting.key, e.target.value)}
+        />
+      );
+    }
+
+    case 'boolean': {
+      return (
+        <SelectWithPlaceholder
+          placeholder={setting.placeholder}
+          options={[{ text: 'true' }, { text: 'false' }]}
+          value={value}
+          onChange={e => onChange(setting.key, e.target.value)}
+        />
+      );
+    }
+
+    case 'amountAndUnit': {
+      const { amount, unit } = amountAndUnitToObject(value ?? '');
```

Comment:
I'm not attached to the terminology `amount` and `unit` but couldn't come up with anything better. 
Example: 100mb. `100` is the amount and `mb` is the unit